Sources/AlgoliaSearch-Client/AbstractClient.swift

```diff
@@ -376,7 +387,14 @@ internal struct HostStatus {
                 finalURLParameters![key] = value
             }
         }
-        let request = Request(client: self, method: method, hosts: hostnames, firstHostIndex: 0, path: path, urlParameters: finalURLParameters, headers: headers, jsonBody: body, timeout: currentTimeout, completion:  completion)
+      
+        var tBody = body
+      
+        // If API key is too big, send it in the request's body (if applicable).
+        if let apiKey = _apiKey, tBody != nil, apiKey.count >= _maxAPIKeyLength {
+          tBody!["apiKey"] = apiKey
```

Comment:
nit: `"apiKey"` can be extracted to a constant, like you did with `AbstractClient.xAlgoliaAPIKey`.